brd-ios/breadwallet/src/Redesign/Scenes/Swap/Validation/FailureViewController.swift

```diff
@@ -8,20 +8,84 @@
 //  See the LICENSE file at the project root for license information.
 //
 
+// currently not used, but if we need to, we can expand the VC with this protocol instead of enum directly
+protocol SimpleMessage {
+    var iconName: String { get }
+    var title: String { get }
+    var description: String { get }
+    var firstButtonTitle: String? { get }
+    var secondButtonTitle: String? { get }
+}
+
+enum FailureReason: SimpleMessage {
+    case buy
+    case swap
+    
+    var iconName: String {
+        return "failedStatusIcon"
+    }
+    
+    var title: String {
+        switch self {
+        case .buy:
+            return "There was an error while processing your payment"
+            
+        case .swap:
+            return "There was an error while processing your transaction"
+        }
+    }
+    
+    var description: String {
+        switch self {
+        case .buy:
+            return "Please contact your card issuer/bank or try again with a different payment method."
+            
+        case .swap:
+            return "Please try swapping again or come back later."
+        }
+    }
+    
+    var firstButtonTitle: String? {
+        switch self {
+        case .buy:
+            return "Try a different payment method"
+            
+        case .swap:
+            return "Swap again"
+        }
+    }
+    
+    var secondButtonTitle: String? {
+        switch self {
+        case .buy:
+            return "Contact support"
+            
+        case .swap:
+            return "Back to Home"
+        }
+    }
+}
+
 import UIKit
 
 extension Scenes {
-    static let Failure = FailureViewController.self
+    static var Failure = FailureViewController.self
```

Comment:
why var

Reply:
was trying smth.. but it didnt work.. will revert